test/parallel/test-fs-whatwg-url.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs');
+const os = require('os');
+const URL = require('url').URL;
+const Buffer = require('buffer').Buffer;
+
+function pathToFileURL(p) {
+  if (!path.isAbsolute(p))
```

Comment:
Not entirely necessary but it's a good fallback check just to be safe. If carries a slight performance hit but in a test that is harmless.